cromwell_tools/_cromwell_api.py

```diff
@@ -0,0 +1,425 @@
+            'submission': '2018-01-01T00:01:01.410150Z',
+            'start': '2018-01-01T01:01:01.410150Z',
+            'end': '2018-01-01T02:01:01.410150Z',
+            'name': ['WorkflowName1', 'WorkflowName2'],
+            'additionalQueryResultFields': ['labels', 'parentWorkflowId'],
+            'includeSubworkflows': True
+        }
+        ```
+
+        which will be converted to the following query parameters:
+
+        ```
+        query_params = [
+            {'label': 'cromwell-workflow-id:cromwell-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'},
+            {'status': 'Running'},
+            {'status': 'Succeeded'},
+            {'id': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'},
+            {'additionalQueryResultFields': 'labels'},
+            {'submission': '2018-01-01T00:01:01.410150Z'},
+            {'start': '2018-01-01T01:01:01.410150Z'},
+            {'end': '2018-01-01T02:01:01.410150Z'},
+            {'name': 'WorkflowName1'},
+            {'name': 'WorkflowName2'},
+            {'additionalQueryResultFields': 'labels'},
+            {'additionalQueryResultFields': 'parentWorkflowId'},
+            {'includeSubworkflows': 'true'}
+        ]
+        ```
+        Args:
+            query_dict (dict): A dictionary representing the query key-value paris. The keys should be accepted by the
+                Cromwell or they will get ignored. The values could be str, list or dict.
+
+        Raises:
+            TypeError: If the input query_dict is not a dictionary.
+            ValueError: If a list of values are assigned to a query key that belongs to _cromwell_exclusive_query_keys.
+        Returns:
+            query_params (List[dict]): A composed list of query objects.
+        """
+        if not isinstance(query_dict, dict):
+            raise TypeError('A valid dictionary with query keys is required!')
+
+        query_params = []
+        for k, v in query_dict.items():
+            if k in cls._cromwell_query_keys:
+                if k is 'label' and isinstance(v, dict):
+                    query_params.extend([
+                        {'label': label_key + ':' + label_value} for label_key, label_value in v.items()
+                    ])
+                elif isinstance(v, list):
+                    if k in cls._cromwell_exclusive_query_keys:
+                        raise ValueError('{} cannot be specified multiple times!'.format(k))
+                    query_params.extend([
+                        {k: json.dumps(val)} if not isinstance(val, str) else {k: val} for val in set(v)
+                    ])
+                else:
+                    query_params.append(
+                            {k: json.dumps(v)} if not isinstance(v, str) else {k: v}
+                    )
+            else:
+                logger.info('{} is not an allowed query key in Cromwell, will be ignored in this query.'.format(k))
+        return query_params
+
+    @staticmethod
+    def _parse_status(response):
```

Comment:
How about renaming this to `_parse_workflow_status`? At first I thought it referred to the HTTP status code

Reply:
👍